src/AuthenticationService.php

```diff
@@ -31,7 +32,7 @@
 /**
  * Authentication Service
  */
-class AuthenticationService implements AuthenticationServiceInterface
+class AuthenticationService implements AuthenticationServiceInterface, ImpersonationInterface
```

Comment:
Implementing `ImpersonationInterface` is not necessary since `AuthenticationServiceInterface` already extends it.